packages/palette/src/elements/Tooltip/Tooltip.tsx

```diff
@@ -98,46 +50,43 @@ export const Tooltip: React.FC<TooltipProps> = ({
     v3: { variant: "xs" as TextVariant },
   })
 
+  usePosition({
+    anchorRef,
+    tooltipRef,
+    position: placement,
+    offset: 10,
+    active: visible ?? active,
+  })
+
   return (
-    <Box
-      tabIndex={1}
-      onClick={handleClick}
-      onMouseOver={activate}
-      onMouseOut={deactivate}
-      onFocus={activate}
-      onBlur={deactivate}
-      position="relative"
-      // TODO: Should avoid making assumptions about display
-      display="inline-block"
-      {...rest}
-    >
+    <>
+      {React.cloneElement(children, {
+        ref: anchorRef,
+        tabIndex: 1,
+        onClick: handleClick,
+        onMouseOver: activate,
+        onMouseOut: deactivate,
+        onFocus: activate,
+        onBlur: deactivate,
+      })}
+
       <Tip
         p={size === "sm" ? 0.5 : 2}
         width={width}
         bg="white100"
+        ref={tooltipRef as any}
+        zIndex={1}
         {...(visible
           ? // If there's a visible prop being passed; use that
             { opacity: visible ? 1 : 0 }
           : // Otherwise use the active state
             { opacity: active ? 1 : 0 })}
-        // Positioning
-        {...{
-          bottom: { top: "100%", mt: 0.5 },
-          top: { bottom: "100%", mb: 0.5 },
-        }[placement]}
```

Comment:
They can be very useful in places where a switch statement is inappropriate. This one is a little goofy since it's only two choices. But it's gone now anyway.

Reply:
Accessing objects like this is cool! Just thinking something like the below is a little more boring / readable. I don't know why but any time i see the anonymous form (coupled with jsx and other spread statements) if feels somewhat obscure. 

```tsx
const position = {
  bottom: { top: "100%", mt: 0.5 },
  top: { bottom: "100%", mb: 0.5 },
}

return (
  <Box {...position[placement]} />
)
```

Reply:
Ah, no doubt 👍